pooch/utils.py

```diff
@@ -160,3 +160,36 @@ def make_registry(directory, output, recursive=True):
             # Only use Unix separators for the registry so that we don't go insane
             # dealing with file paths.
             outfile.write("{} {}\n".format(fname.replace("\\", "/"), fhash))
+
+
+def parse_url(url):
+    """
+    Parse a URL into 3 components:
+
+    <protocol>://<netloc>/<path>
+
+    Parameters
+    ----------
+    url : str
+        URL (e.g.: http://127.0.0.1:8080/test.nc, ftp://127.0.0.1:8080/test.nc)
+
+    Returns
+    -------
+    parsed_url : dict
+        Three components of a URL
+        (e.g., {'protocol': 'http', 'netloc': '127.0.0.1:8080', 'path': '/test.nc'})
+
+    Examples
+    --------
+
```

Comment:
Not sure how to make the  following doctest failure go away

```python
=================================== FAILURES ===================================
_______________________ [doctest] pooch.utils.parse_url ________________________
176     Returns
177     -------
178     parsed_url : dict
179         Three components of a URL
180         (e.g., {'protocol': 'http', 'netloc': '127.0.0.1:8080', 'path': '/test.nc'})
181 
182     Examples
183     --------
184 
185     >>> print(parse_url("http://127.0.0.1:8080/test.nc"))
Expected:
    {'protocol': 'http', 'netloc': '127.0.0.1:8080', 'path': '/test.nc'}
Got:
    {'protocol': 'http', 'path': '/test.nc', 'netloc': '127.0.0.1:8080'}

```

Reply:
The problem here is that dictionaries aren't always ordered. So it's not good to use them in doctests (unless you go through the trouble of sorting them).